internal/querycoordv2/checkers/segment_checker.go

```diff
@@ -282,26 +285,31 @@ func (c *SegmentChecker) getSealedSegmentDiff(
 	nextTargetExist := c.targetMgr.IsNextTargetExist(ctx, collectionID)
 	nextTargetMap := c.targetMgr.GetSealedSegmentsByCollection(ctx, collectionID, meta.NextTarget)
 	currentTargetMap := c.targetMgr.GetSealedSegmentsByCollection(ctx, collectionID, meta.CurrentTarget)
-
```

Comment:
finishRecover is true when:
1. at least one of current and next target is existed, this is to avoid the case where the querynode just started and neither of these two targets are present
2. none of segments existed in the current target is absent
3. none of segments existed in the next target of a recovering replica
all these three conditions are met, then finishRecover is true